mlflow/pyspark/ml/__init__.py

```diff
@@ -925,8 +939,56 @@ def _log_posttraining_metadata(estimator, spark_model, params):
         if log_models:
             if _should_log_model(spark_model):
                 # TODO: support model signature
+                from mlflow.models.signature import infer_signature
+
+                def _cast_spark_df_with_vector_to_array(input_spark_df):
+                    """
+                    Finds columns of vector type in a spark dataframe and
+                    casts them to array<double> type.
+
+                    :param input_spark_df:
+                    :return: a spark dataframe with vector columns transformed to array<double> type
+                    """
+                    from functools import reduce
+                    from pyspark.ml.functions import vector_to_array
+                    from pyspark.ml.linalg import VectorUDT
+
+                    vector_type_columns = [
+                        _field.name
+                        for _field in input_spark_df.schema
+                        if isinstance(_field.dataType, VectorUDT)
+                    ]
+                    return reduce(
+                        lambda df, vector_col: df.withColumn(
+                            vector_col, vector_to_array(vector_col)
+                        ),
+                        vector_type_columns,
+                        input_spark_df,
+                    )
+
+                limited_input_df = input_df.limit(INPUT_EXAMPLE_SAMPLE_ROWS)
```

Comment:
My concern is the collecting example rows will trigger one or more spark job.
Suppose the spark dataframe haven't been cached and the spark dataframe contains a long lineage, then it might generate a bunch of spark jobs.
and when we do training, these spark jobs might run again.

Reply:
so shall we make the signature logging enabled by default ?

Reply:
@WeichenXu123, that is a great point! I noticed many of the spark estimators handle persistence internally by persisting before the fit and unpersisting afterward. Is it OK if we do it upfront (`input_df.persist(StorageLevel.MEMORY_AND_DISK)`) before the original fit is called and then unpersist it after autologging ?

Reply:
> handle persistence internally by persisting before the fit and unpersisting afterward

This sounds a good idea.